runtime/bcverify/staticverify.c

```diff
@@ -1322,25 +1317,31 @@ checkStackMap (J9CfrClassFile* classfile, J9CfrMethod * method, J9CfrAttributeCo
 					}
 					if (frameType >= CFR_STACKMAP_CHOP_3) {
 						slotCount = (IDATA) frameType - CFR_STACKMAP_APPEND_BASE;
-						localSlots += slotCount;
 						if (slotCount < 0) {
 							slotCount = 0;
 						}
+
+						if ((frameType >= CFR_STACKMAP_APPEND_1) && (frameType <= CFR_STACKMAP_APPEND_3)) {
+							maxForEntries = code->maxLocals;
+						}
 					}
-			
 				} else {
 					/* full frame */
 					/* Executed with StackMap or StackMapTable */
 					NEXT_U16(slotCount, entries);
-					localSlots = slotCount;
-					errorCode = checkStackMapEntries (classfile, code, map, &entries, slotCount, end);
+
+					errorCode = checkStackMapEntries (classfile, code, map, &entries, slotCount, end, code->maxLocals);
 					if (0 != errorCode) {
 						goto _failedCheck;
 					}
 					NEXT_U16(slotCount, entries);
+					/* The stackmap entry is invalid if the size of stack (1 slot) exceeds the size of the max stack.*/
+					if (code->maxStack < slotCount) {
+						errorCode = FATAL_CLASS_FORMAT_ERROR;
```

Comment:
this is incorrect as slotCount reflects the current slot count from 0, which means it only contain the count of `locals `rather than `stack`.

The whole stackframe looks like:
```
[ localSlot1, localSlot2, .... | stackSlot1, stackSlot2....] 
           ^ ------------------------------->
           slotCount <---- it is used to record the location of current slot
```